public/app/features/alerting/unified/components/rule-viewer/AlertRuleMenu.tsx

```diff
@@ -162,14 +180,19 @@ const AlertRuleMenu = ({
           ))}
         </>
       )}
-      {canDelete && rulerRule && (
+      {canDelete && (
         <>
           <Menu.Divider />
           <Menu.Item
             label={t('alerting.common.delete', 'Delete')}
             icon="trash-alt"
             destructive
-            onClick={() => handleDelete(rulerRule, groupIdentifier)}
+            onClick={() => {
+              // if the identifier is not for a editable rule I wonder how you even got here.
+              if (isEditableRuleIdentifier(identifier)) {
```

Comment:
required to pass TypeScript type checking because this menu works with any `RuleIdentifier` and not just `EditableRuleIdentifier`